src/jit/compiler.h

```diff
@@ -5410,6 +5410,14 @@ class Compiler
     CSEdsc**            optCSEhash;
     CSEdsc**            optCSEtab;
 
+    typedef SimplerHashTable<GenTreePtr, PtrKeyFuncs<GenTree>, GenTreePtr, JitSimplerHashBehavior> NodeToNodeMap;
+
+    NodeToNodeMap* cseArrLenMap; // Maps array length nodes to ancestor compares that should be
+                                 // re-numbered with the array length to improve range check elimination
+
+    // Given a compare, look for a cse candidate arrlen feeding it and add a map entry if found.
+    void updateCseArrLenMap(GenTreePtr compare);
```

Comment:
All of the other CSE related methods defined in compiler.h do start with an opt prefix, so it probably best to continue to do so here.